src/game/TileEngine/SmokeEffects.cc

```diff
@@ -196,7 +196,7 @@ void NewSmokeEffect(const INT16 sGridNo, const UINT16 usItem, const INT8 bLevel,
 	pSmoke->ubRadius    = ubStartRadius;
 	pSmoke->bAge				= 0;
 	pSmoke->fAllocated  = TRUE;
-	pSmoke->bType				= bSmokeEffectType;
+	pSmoke->bType				= static_cast<int8_t>(smokeEffectID);
```

Comment:
Looks like SMOKEEFFET::bType is not actually used for anything so you could just remove it.

Reply:
True. But it should be there for savegame compatibility. I think with the next PR i might be able to read it from the explosive item though and can remove it from the struct.